### ISSUE TITLE
Branches names containing `/` in are not supported

### ISSUE THREAD

**Issue by cheind** [jupyter/nbviewer#665]
Just figured that branches containing a slash such as `feature/x` are not treated correctly by nbviewer.  To reproduce go to 

http://nbviewer.jupyter.org/github/cheind/py-cgraph/tree/master/

and try to switch to 'feature/sdf_base'. Frontend will respond with `404 : Not Found`.  Branch names containing slashes are not uncommon when you work with git-flow. A quick fix involves escaping `/` by `%2F`. 

Instead of `http://nbviewer.jupyter.org/github/cheind/py-cgraph/tree/feature/sdf_base/`
try `http://nbviewer.jupyter.org/github/cheind/py-cgraph/tree/feature%2Fsdf_base/`

**Comment by minrk** [comment]
Indeed, we should make sure that these work. I would require that the branch name be escaped so it's a single URL part. We should just make sure that the links, etc. do this correctly everywhere.
